### ISSUE TITLE
Consider RELAYHOST during the SPF generation

### ISSUE THREAD

**Issue by gmelillo** [Mailu/Mailu#2269]
## Before you open your issue
- [X] Check if no issue or pull-request for this already exists.
- [X] Check [documentation](https://mailu.io/master/) and [FAQ](https://mailu.io/master/faq.html). (Tip, use the search function on the documentation page)
- [X] You understand `Mailu` is made by volunteers in their **free time** — be conscise, civil and accept that delays can occur.
- [ ] The title of the issue should be short and simple. It should contain specific terms related to the actual issue. Be specific while writing the title.

### Environment
 - [ ] docker-compose
 - [X] kubernetes
 - [ ] docker swarm

### Versions

1.9.20

## Description
In my current setup I've configured a `relayhost` to route all outgoing email using a cloud server.
This configuration is working correctly but in the domain details I still see:

```
example.com. 600 IN TXT "v=spf1 mx a:mail.example.com ~all"
```

## Replication Steps
- Install mailu with `external_relay.host` configured

## Expected behaviour
If I do configure an IP address or host in `external_relay.host` I would expect to see a change in the SPF record.

Example IP `5.5.5.5`:
```
example.com. 600 IN TXT "v=spf1 mx a:mail.example.com ip4:5.5.5.5 ~all"
```

Example IP `fe80::1%lo0`:
```
example.com. 600 IN TXT "v=spf1 mx a:mail.example.com ip6:fe80::1%lo0 ~all"
```

Example host `relay.example.com`:
(where it resolve 1 ipv4 and 1 ipv6)
```
example.com. 600 IN TXT "v=spf1 mx a:mail.example.com ip4:5.5.5.5 ip6:fe80::1%lo0 ~all"
```

If that type of change is welcome I can also start to work on a PR to propose this behaviour change.

**Comment by Diman0** [comment]
I think you are referring to the RELAYHOST setting (https://mailu.io/1.9/configuration.html#mail-settings)?
Using this will indeed not lead to updated SPF records. I will mark this as an enhancement request.

**Comment by nextgens** [comment]
IMHO we shouldn't propose an SPF record when a relayhost is configured. The delivery chain may be much more complicated and assuming that the IP of the relay will be the final one is very naive.

**Comment by gmelillo** [comment]
@Diman0 Yes, I'm referring on that one. Also @nextgens have a point there. There is no way to know if the relay host is the final point that relay the mail.

To mitigate this a new config can be introduced to specify the additions to the SPF. That way also complicated routing can be covered.

**Comment by Diman0** [comment]
Nextgens indeed has a point. Introducing yet another configuration variable for this makes the solution too complex without valid reason. If you have the information to add to this new variable, then you can already add it directly to your DNS config. No need to add it to a config, read the exact same information from Mailu, and then add it to your DNS config.
